### ISSUE TITLE
Ray caster (height scanner) for terrains higher than the robot

### ISSUE THREAD

**Issue by Wang-Haitong** [mujocolab/mjlab#775]
Hi,

The current setup of height scanner in the velocity tasks does not work for terrains higher than the robot. For example, if a quadruped is climbing a box that's higher than its standing height. In IsaacLab, an offset is added to the height scanner (often set to 20m in z) so that it can measure high terrains. However, if adding a site to robot xml that has a z set to 20m, yaw alignment would "fail". Because any slight movement of the base (e.g., small pitch) would cause the ray origin (20m tall) move a lot in x axis.

My current fix is to add an offset to the GridPatternCfg, it seems to work.

```
@dataclass
class OffsetGridPatternCfg(GridPatternCfg):

  offset: tuple[float, float, float] = (0.0, 0.0, 0.0)
  """Positional offset (x, y, z) added to every ray origin in local frame."""

  def generate_rays(
    self, mj_model: mujoco.MjModel | None, device: str
  ) -> tuple[torch.Tensor, torch.Tensor]:
    offsets, directions = super().generate_rays(mj_model, device)
    offsets += torch.tensor(self.offset, device=device, dtype=torch.float32)
    return offsets, directions
```

**Comment by kevinzakka** [comment]
Hi @Wang-Haitong, this has been fixed in #778. Mind testing it and letting me know if it solves your problem?

**Comment by Wang-Haitong** [comment]
Hi @kevinzakka it works now, thank you!
